src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/ISimdVector_2.cs

```diff
@@ -0,0 +1,562 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System.Runtime.Intrinsics
+{
+    /// <summary>Defines a single instruction, multiple data (SIMD) vector type.</summary>
+    /// <typeparam name="TSelf">The type that implements the interface.</typeparam>
+    /// <typeparam name="T">The type of the elements in the vector.</typeparam>
+    internal unsafe interface ISimdVector<TSelf, T>
+        : IAdditionOperators<TSelf, TSelf, TSelf>,
+       // IAdditiveIdentity<TSelf, TSelf>,
+          IBitwiseOperators<TSelf, TSelf, TSelf>,
+       // IComparisonOperators<TSelf, TSelf, bool>,
+       // IDecrementOperators<TSelf>,
+          IDivisionOperators<TSelf, TSelf, TSelf>,
+          IEqualityOperators<TSelf, TSelf, bool>,
+          IEquatable<TSelf>,
+       // IIncrementOperators<TSelf>,
+       // IMinMaxValue<TSelf>,
+       // IModulusOperators<TSelf, TSelf, TSelf>,
+       // IMultiplicativeIdentity<TSelf, TSelf>,
+          IMultiplyOperators<TSelf, TSelf, TSelf>,
+          IShiftOperators<TSelf, int, TSelf>,
+       // ISpanFormattable,
+          ISubtractionOperators<TSelf, TSelf, TSelf>,
+          IUnaryNegationOperators<TSelf, TSelf>,
+          IUnaryPlusOperators<TSelf, TSelf>
+       // IUtf8SpanFormattable
+       where TSelf : ISimdVector<TSelf, T>?
+    {
+        /// <summary>Gets the natural alignment of the vector.</summary>
+        /// <exception cref="NotSupportedException">The type of the elements in the vector (<typeparamref name="T" />) is not supported.</exception>
+        static abstract int Alignment { get; }
+
+        /// <summary>Gets an instance of the vector type in which all bits are set.</summary>
+        /// <exception cref="NotSupportedException">The type of the elements in the vector (<typeparamref name="T" />) is not supported.</exception>
+        static abstract TSelf AllBitsSet { get; }
+
+        /// <summary>Gets the number of <typeparamref name="T" /> that are in the vector.</summary>
+        /// <exception cref="NotSupportedException">The type of the elements in the vector (<typeparamref name="T" />) is not supported.</exception>
+        static abstract int Count { get; }
+
+        /// <summary>Gets a value that indicates whether the vector operations are subject to hardware acceleration through JIT intrinsic support.</summary>
+        /// <value><see langword="true" /> if the vector operations are subject to hardware acceleration; otherwise, <see langword="false" />.</value>
```

Comment:
Can we inherit most of these docs rather than duplicating them?

Reply:
Ideally most things would eventually inherit their docs from the interface.

I'd prefer to get this in (so we can get perf numbers) and log an issue tracking us change the `Vector128<T>` and other types to inherit from here instead in a follow up PR.